src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc

```diff
@@ -300,7 +421,7 @@ BtreeLBAManager::base_iertr::future<> _init_cached_extent(
       LOG_PREFIX(BtreeLBAManager::init_cached_extent);
       if (!iter.is_end() &&
 	  iter.get_key() == logn->get_laddr() &&
-	  iter.get_val().paddr == logn->get_paddr()) {
+	  (paddr_t)iter.get_val().pladdr == logn->get_paddr()) {
```

Comment:
iter.get_val().pladdr must be a physical one, because the laddr given by `logn->get_laddr()` will always be the laddr of the `origianl data mapping`. There is an assert in `pladdr_t::paddr_t()` that assure this.

Reply:
I think the problem is that the input extent `e/logn` may be outdated/wrong, the logic here is to mark this case as false, not to abort the program with assert failure, right?

Reply:
> I think the problem is that the input extent `e/logn` may be outdated/wrong, the logic here is to mark this case as false, not to abort the program with assert failure, right?

Yes, but outdated/wrong `e/logn`s' pladdr must still be a physical one, it can't be a logical one. This pr holds the invariant that all logical extents always have the keys of the original data mappings as their laddr.